src/AutoRest.CSharp/Common/Output/Builders/ClientBuilder.cs

```diff
@@ -90,11 +92,13 @@ public static IEnumerable<PagingMethod> BuildPagingMethods(OperationGroup operat
                         throw new InvalidOperationException($"Method {method.Name} has to have a return value");
                     }
 
+                    var name = nameOverrider?.Invoke(operationGroup, operation, method) ?? method.Name;
+
                     yield return new PagingMethod(
                         method,
                         nextPageMethod,
-                        method.Name,
-                        new Diagnostic($"{Declaration.Name}.{method.Name}"),
+                        name,
+                        new Diagnostic($"{Declaration.Name}.{name}"),
```

Comment:
We make this change to ensure the Diagnostic scope is correct if we have overridden the name of the method